ansible_collections/arista/avd/plugins/plugin_utils/cv_client/client/studio.py

```diff
@@ -40,6 +50,79 @@ class StudioMixin:
 
     studio_api_version: Literal["v1"] = "v1"
 
+    async def get_studio(
+        self: CVClient,
+        studio_id: str,
+        workspace_id: str,
+        time: datetime | None = None,
+        timeout: float = 10.0,
+    ) -> Studio:
+        """
+        Get Studio definition using arista.studio.v1.StudioService.GetOne.
+
+        The Studio GetOne API for the workspace does not return anything from mainline and does not return deletions in the workspace.
+        So to produce the Workspace Studio we need to fetch from the workspace, and if we find nothing, we need to check if the studio
+        got deleted in the workspace by retrieving config. Finally we can fetch from mainline.
+
+        Parameters:
+            studio_id: Unique identifier for the studio.
+            workspace_id: Unique identifier of the Workspace for which the information is fetched. Use "" for mainline.
+            time: Timestamp from which the information is fetched. `now()` if not set.
+            timeout: Timeout in seconds.
+
+        Returns:
+            Studio object.
+        """
+        request = StudioRequest(
+            # First attempt to fetch studio from workspace.
+            key=StudioKey(studio_id=studio_id, workspace_id=workspace_id),
+            time=time,
+        )
+        client = StudioServiceStub(self._channel)
+        try:
+            response = await client.get_one(request, metadata=self._metadata, timeout=timeout)
+            return response.value
+        except Exception as e:
+            e = get_cv_client_exception(e, f"Studio ID '{studio_id}, Workspace ID '{workspace_id}'") or e
+            if isinstance(e, CVResourceNotFound):
+                # Continue execution if we did not find any state in the workspace.
+                # This simply means the studio itself was not changed in this workspace.
+                pass
+            else:
+                raise e
+
+        # If we get here, it means no studio was returned by the workspace call.
+        # So now we fetch the studio config from the workspace to see if the studio was deleted in this workspace.
+        request = StudioConfigStreamRequest(
```

Comment:
Is this working?

Reply:
I am not familiar with the wrappers you have implemented, pretty neat. It's a hassle to deal with the proto gen python clients.

 But the first argument needs to be list.

Reply:
fixed. Thanks